csrc/moe/topk_softmax_kernels.cu

```diff
@@ -383,7 +390,7 @@ struct TopkConstants
 {
     static constexpr int ELTS_PER_LDG = BYTES_PER_LDG / sizeof(float);
     static_assert(EXPERTS / (ELTS_PER_LDG * WARP_SIZE) == 0 || EXPERTS % (ELTS_PER_LDG * WARP_SIZE) == 0, "");
-    static constexpr int VECs_PER_THREAD = std::max(1, EXPERTS / (ELTS_PER_LDG * WARP_SIZE));
+    static constexpr int VECs_PER_THREAD = MAX(1, EXPERTS / (ELTS_PER_LDG * WARP_SIZE));
```

Comment:
Why though?

Reply:
The hip compiler had a tough time using std::max to create a constexpr variable.
![image](https://github.com/ROCm/vllm/assets/137818590/d591a0ec-27d6-4399-a98f-54d7ce36d71d)

Reply:
You're right. Such a shame.
We may want to submit a bug to hipclang for not complying with c++14 in /opt/rocm-6.1.0/lib/llvm/lib/clang/17/include/__clang_hip_math.h
I mean, it's only been 10 years, right?